src/utils/line.ts

```diff
@@ -51,54 +58,138 @@ export const formatCreditEvents = (
+};
+
 /**
  * @function
  * @name mergeCollateralEvents
- * @desc - takes all events for a single deposit/spigot and merges them into global list
+ * @desc - takes all events for a single escrow deposit/spigot and merges them into global list
  * @dev  - expects all events to be in the same token
  * @param type - the type of module used as collateral
  * @param symbol - the token in event
- * @param price - current price for escrow and spigot collateral
+ * @param price - current price for escrow deposit/spigot collateral
  * @param events - the events to process
  * @return totalVal, CollateralEvent[] - current total value of all collateral
  */
+// export const formatCollateralEvents = (
+//   type: ModuleNames,
+//   symbol: string,
+//   price: BigNumber = BigNumber.from(0),
+//   // events: EscrowDepositList,
+//   // events: CollateralEvent[],
+//   events: EscrowEventFragResponse[] | undefined,
+//   // events: EscrowDepositList, //| SpigotEvents, //
+//   tokenRevenue?: any
+// ): [number, CollateralEvent[]] => {
+//   let totalVal = 0;
+//   if (!events) return [totalVal, []];
+//   // TODO promise.all token price fetching for better performance
+//   const newEvents: (CollateralEvent | undefined)[] = events?.map((event: any): CollateralEvent | undefined => {
+//     console.log('get line page data - individual event: ', event);
+//     const { __typename, timestamp, amount, token, value = unnullify(0, true) } = event;
+//     if (!timestamp || !amount) return undefined;
+//     console.log('get line page data - individual event 2: ', token);
+//     const valueNow = unnullify(price.toString(), true).times(unnullify((amount.toString(), true)));
+//     let collatType;
+//     switch (type) {
+//       case SPIGOT_MODULE_NAME:
+//         // aggregate token revenue. not needed for escrow bc its already segmented by token
+//         // use price at time of revenue for more accuracy
+//         tokenRevenue[symbol] += parseUnits(unnullify(tokenRevenue[symbol], true), 'ether').add(value).toString();
+//         collatType = COLLATERAL_TYPE_REVENUE;
+//         break;
+//       case ESCROW_MODULE_NAME:
+//         collatType = COLLATERAL_TYPE_ASSET;
+//         break;
+//       default:
+//         break;
+//     }
+
+//     totalVal += valueNow;
+
+//     return {
+//       type: collatType,
+//       __typename,
+//       timestamp,
+//       amount,
+//       value,
+//       valueNow,
+//       id: token?.id,
+//     };
+//   });
+//   const validEvents = newEvents.filter((x) => !!x) as CollateralEvent[];
+//   console.log('line page data - valid events: ', validEvents);
+//   return [totalVal, validEvents];
+// };
+
 export const formatCollateralEvents = (
   type: ModuleNames,
-  symbol: string,
+  token: TokenFragRepsonse,
   price: BigNumber = BigNumber.from(0),
-  events: CollateralEvent[],
+  // events: EscrowDepositList,
+  // events: CollateralEvent[],
+  events: EscrowEventFragResponse[] | undefined,
```

Comment:
Also needs to handle Spigot events

Reply:
I don't think this is ideal, but for now I've created a separate function to handle Spigot events, `formatSpigotCollateralEvents`. 

The `formatCollateralEvents` function is currently used for Escrow events. As I'm sure you are well aware, Spigot Events and Escrow events have different schemas / response types so separate logic is needed to handle them. This can be done with 2 functions or if/else logic within `formatCollateralEvents`.